server/models/Transaction.js

```diff
@@ -262,10 +262,14 @@ function defineModel() {
       },
 
       hooks: {
-        afterCreate: transaction => {
-          Transaction.createActivity(transaction);
-          // intentionally returns null, needs to be async (https://github.com/petkaantonov/bluebird/blob/master/docs/docs/warning-explanations.md#warning-a-promise-was-created-in-a-handler-but-was-not-returned-from-it)
-          return null;
+        afterCreate: (transaction, options) => {
+          if (options?.transaction) {
+            return Transaction.createActivity(transaction, { transaction: options.transaction });
```

Comment:
My personal experience with working on that in https://github.com/opencollective/opencollective-api/pull/5981 is that this will not play well in some cases because `createActivity` then gets null results when it tries to fetch the transaction and its associations by ID. If it's not essential for this PR, I would prefer to wait for a proper revamp of this code.

Reply:
My issue is that I want to create expenses and transactions within the same postgres transaction in case it fails and `Transaction.createActivity` is async.
Being async is not a problem when you're not operating within a postgres transaction, but if you are, it means that both Transaction and Activity need to be awaited for (that's why I'm also returning the `createActivity`) here.

The alternative is not using a postgres transaction in the function that processes privacy transactions, it would necessary means that we need to manually roll back created instances if things go wrong.